plugins/modules/discovery_intent.py

```diff
@@ -200,6 +200,101 @@
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for CLI authentication, mandatory when using global CLI credentials.
+                        type: str
+                    description:
+                        description: Name of the CLI credential, mandatory when using global CLI credentials.
+                        type: str
+            http_read_credential_list:
+                description:
+                    - List of Global HTTP Read credentials to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique username or description.
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for HTTP Read authentication, mandatory when using global HTTP credentials.
+                        type: str
+                    description:
+                        description: Name of the HTTP Read credential, mandatory when using  global HTTP credentials.
+                        type: str
+            http_write_credential_list:
+                description:
+                    - List of Global HTTP Write credentials to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique username or description.
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for HTTP Write authentication, mandatory when using global HTTP credentials.
+                        type: str
+                    description:
+                        description: Name of the HTTP Write credential, mandatory when using  global HTTP credentials.
+                        type: str
+            snmp_v2_read_credential_list:
+                description:
+                    - List of Global SNMP V2 Read credentials to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique username or description.
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for SNMP Read authentication, mandatory when using global SNMP credentials.
+                        type: str
+                    description:
+                        description: Name of the SNMP Read credential, mandatory when using  global SNMP credentials.
+                        type: str
+            snmp_v2_write_credential_list:
+                description:
+                    - List of Global SNMP V2 Write credentials to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique username or description.
```

Comment:
`It's recommended to create device credentials with both a unique username and a clear description for easy identification`